api-service/src/main/java/com/cyverse/api/services/AuthService.java

```diff
@@ -0,0 +1,85 @@
+package com.cyverse.api.services;
+
+import com.auth0.jwt.JWT;
+import com.auth0.jwt.JWTVerifier;
+import com.auth0.jwt.algorithms.Algorithm;
+import com.auth0.jwt.exceptions.JWTVerificationException;
+import com.auth0.jwt.interfaces.DecodedJWT;
+import com.cyverse.api.config.AuthUserConfig;
+import com.cyverse.api.exceptions.ApiTokenExpiredException;
+import com.cyverse.api.exceptions.UnauthorizedAccessException;
+
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Authentication service based on JWTs.
+ */
+public class AuthService {
+
+    private final Map<String, AuthUserConfig> users;
+    private static final String TOKEN_ISSUER = "http://api-service.cyverse.at";
```

Comment:
This should be configurable.
```suggestion
    private static final String TOKEN_ISSUER = "http://api-service.cyverse.at";
```